cuqi/experimental/mcmc/_cwmh.py

```diff
@@ -0,0 +1,207 @@
+    sampling of a proposal distribution along with an accept/reject step.
+
+    Parameters
+    ----------
+
+    target : `cuqi.distribution.Distribution` or lambda function
+        The target distribution to sample. Custom logpdfs are supported by using
+        a :class:`cuqi.distribution.UserDefinedDistribution`.
+    
+    proposal : `cuqi.distribution.Distribution` or callable method
+        The proposal to sample from. If a callable method it should provide a
+        single independent sample from proposal distribution. Defaults to a
+        Gaussian proposal.  *Optional*.
+
+    scale : float or ndarray
+        Scale parameter used to define correlation between previous and proposed
+        sample in random-walk.  *Optional*. If float, the same scale is used for
+        all dimensions. If ndarray, a (possibly) different scale is used for
+        each dimension.
+
+    initial_point : ndarray
+        Initial parameters. *Optional*
+
+    dim : int
+        Dimension of parameter space. Required if target and proposal are
+        callable functions. *Optional*.
+
+    callback : callable, *Optional*
+        If set this function will be called after every sample.
+        The signature of the callback function is
+        `callback(sample, sample_index)`, where `sample` is the current sample
+        and `sample_index` is the index of the sample.
+        An example is shown in demos/demo31_callback.py.
+
+    kwargs : dict
+        Additional keyword arguments to be passed to the base class 
+        :class:`ProposalBasedSamplerNew`.
+
+    Example
+    -------
+    .. code-block:: python
+        import numpy as np
+        import cuqi
+        # Parameters
+        dim = 5 # Dimension of distribution
+        mu = np.arange(dim) # Mean of Gaussian
+        std = 1 # standard deviation of Gaussian
+
+        # Logpdf function
+        logpdf_func = lambda x: -1/(std**2)*np.sum((x-mu)**2)
+
+        # Define distribution from logpdf as UserDefinedDistribution (sample
+        # and gradients also supported as inputs to UserDefinedDistribution)
+        target = cuqi.distribution.UserDefinedDistribution(
+            dim=dim, logpdf_func=logpdf_func)
+
+        # Set up sampler
+        sampler = cuqi.experimental.mcmc.CWMHNew(target, scale=1)
+
+        # Sample
+        samples = sampler.sample(2000).get_samples()
+
+    """
+    def __init__(self, target: cuqi.density.Density, proposal=None, scale=1,
+                 initial_point=None, **kwargs):
+        super().__init__(target, proposal=proposal, scale=scale,
+                         initial_point=initial_point, **kwargs)
+        # Set scale
+        if isinstance(scale, Number):
+            self.scale = np.ones(self.dim)*scale
+            self._scale_temp = np.ones(self.dim)*scale
+        elif isinstance(scale, np.ndarray):
+            self.scale = scale
+            self._scale_temp = scale.copy()
```

Comment:
We may want to include this in scale setter method for robustness?

Reply:
good idea! done